apps/question-the-docs/backend/ai/utils/github.py

```diff
@@ -51,3 +55,29 @@ def save_github_md_files_locally(owner, name, documentation_location):
             f.write(content)
 
     return Path(f"docs/{name}").glob("*")
+
+
+def clone_repo(git_url, target_dir):
```

Comment:
As discussed, my personal preference is to use the web API, as I feel it looks 'cleaner'. Perhaps @blythed can give his opinion here.

Reply:
Yes, looks more professional and the user might not have `.git` installed for some reason.